src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Properties.jl

```diff
@@ -38,8 +38,9 @@ function is_smooth(P::AbsProjectiveScheme; algorithm=:covered)
 end
 
 @attr Bool function is_irreducible(P::AbsProjectiveScheme)
+  is_empty(P) && return false  # the empty set is not irreducible
   I = defining_ideal(P)
-  return is_prime(I)
+  return is_prime(radical(I)) # topological property
```

Comment:
Thanks for catching this! 

But maybe something like `is_primary` exists and is faster? I don't know! But let's ask around.